homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -0,0 +1,141 @@
+    CONF_INTERFACE,
+    CONF_TRY_HOTSPOT,
+    DEFAULT_CONSIDER_HOME,
+    DEFAULT_INTERFACE,
+    DEFAULT_NAME,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TELNET_PORT,
+    DOMAIN,
+)
+
+
+class KeeneticFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return KeeneticOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    return self.async_abort(reason="already_configured")
+                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+                    errors[CONF_NAME] = "name_exists"
+                    break
+
+            try:
+                _client = Client(
+                    TelnetConnection(
+                        user_input[CONF_HOST],
+                        user_input[CONF_PORT],
+                        user_input[CONF_USERNAME],
+                        user_input[CONF_PASSWORD],
+                    )
+                )
+
+                await self.hass.async_add_executor_job(_client.get_router_info)
+            except ConnectionException:
+                errors["base"] = "cannot_connect"
+
+            if not errors:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(CONF_PORT, default=DEFAULT_TELNET_PORT): int,
+                }
+            ),
+            errors=errors,
+        )
+
+
+class KeeneticOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, _user_input=None):
+        """Manage the options."""
+        return await self.async_step_device_tracker()
+
+    async def async_step_device_tracker(self, user_input=None):
+        """Manage the device tracker options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SCAN_INTERVAL,
```

Comment:
It's controlling how often the router will be polled. Changing this value can improve user experience for cases when they want to have a faster reaction or lower network load.

Reply:
Do we know that there are users that need this? We don't want to add features that just potentially will be used.

Reply:
I use it to tune one of my installations to a very extreme strategy along with `consider home` value since I have a robust network connecting Pi to the router and want to catch devices as early as possible. 
But I don't think sub-10 seconds polling is a good default.

I searched thru the codebase, and it's not a rare config option now.

Reply:
Other existing integrations are not relevant for checking if an option is used for an integration. But you say that at least you use this option so we know that there's at least one user for this use case.